pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2,301 +2,1132 @@ package listusers
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type:     "group",
+						Relation: "", // should be "member"
 					},
 				},
 			},
+			model: model,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("group:eng", "member", "user:will"),
+				tuple.NewTupleKey("group:eng", "member", "user:maria"),
+				tuple.NewTupleKey("group:marketing", "member", "user:jon"),
+				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
+			},
+			expectedUsers: []string{},
 		},
 		{
-			name: "direct_relationship_through_multiple_usersets",
+			name:                  "userset_group_granularity_with_direct_user_relationships",
+			TemporarilySkipReason: "because `group:eng` is being returned instead of `group:fga#member`",
 			req: &openfgav1.ListUsersRequest{
-				StoreId:              storeID,
-				AuthorizationModelId: modelID,
-				Object:               &openfgav1.Object{Type: "document", Id: "1"},
-				Relation:             "viewer",
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type:     "group",
+						Relation: "member",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type group
+				relations
+					define member: [user]
+			type document
+				relations
+					define viewer: [ user, group#member ]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("group:eng", "member", "user:will"),
+				tuple.NewTupleKey("group:eng", "member", "user:maria"),
+				tuple.NewTupleKey("group:marketing", "member", "user:jon"),
+				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
+
+				tuple.NewTupleKey("document:1", "viewer", "user:poovam"),
+			},
+			expectedUsers: []string{"group:eng#member"},
+		},
+		{
+			name: "userset_multiple_usersets",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
 				UserFilters: []*openfgav1.ListUsersFilter{
 					{
 						Type: "user",
 					},
 				},
 			},
-			typeDefinitions: parser.MustParse(`
+			model: `model
+            schema 1.1
 			type user
 			type group
 			  relations
-			    define member: [user, group#member] as self
+			    define member: [user, group#member]
 			type document
 			  relations
-			    define viewer: [group#member] as self
-			`),
+			    define viewer: [group#member]`,
 			tuples: []*openfgav1.TupleKey{
-				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
 				tuple.NewTupleKey("group:eng", "member", "user:hawker"),
+				tuple.NewTupleKey("group:fga", "member", "user:jon"),
 				tuple.NewTupleKey("group:eng", "member", "group:fga#member"),
+				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
```

Comment:
Just as a sanity check in addition to the nested group, can you add one more group at the `document:1#viewer` level?

e.g.
* document:1#viewer@group:eng#member
* document:1#viewer@group:other#member
* group:eng#member@group:fga#member
* group:eng#member@user:hawker
* group:fga#member@user:jon
* group:other#member@user:will

expected [`user:jon`, `user:hawker`, `user:will`]